bpfman/src/oci_utils/image_manager.rs

```diff
@@ -611,33 +579,27 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn test_good_image_content_path() {
-        let tmpdir = tempfile::tempdir().unwrap();
-        std::env::set_current_dir(&tmpdir).unwrap();
-
+    async fn test_good_image_content_key() {
         struct Case {
             input: &'static str,
-            output: PathBuf,
+            output: &'static str,
         }
         let tt = vec![
-            Case{input: "busybox", output: tmpdir.as_ref().to_path_buf().join("docker.io/library/busybox/latest")},
-            Case{input:"quay.io/busybox", output: tmpdir.as_ref().to_path_buf().join("quay.io/busybox/latest")},
-            Case{input:"docker.io/test:tag", output: tmpdir.as_ref().to_path_buf().join("docker.io/library/test/tag")},
-            Case{input:"quay.io/test:5000", output: tmpdir.as_ref().to_path_buf().join("quay.io/test/5000")},
-            Case{input:"test.com/repo:tag", output: tmpdir.as_ref().to_path_buf().join("test.com/repo/tag")},
+            Case{input: "busybox", output: "docker.io_library_busybox_latest"},
+            Case{input:"quay.io/busybox", output: "quay.io_busybox_latest"},
+            Case{input:"docker.io/test:tag", output: "docker.io_library_test_tag"},
+            Case{input:"quay.io/test:5000", output: "quay.io_test_5000"},
+            Case{input:"test.com/repo:tag", output: "test.com_repo_tag"},
             Case{
                 input:"test.com/repo@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
-                output: tmpdir.as_ref().to_path_buf().join("test.com/repo/sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"),
+                output: "test.com_repo_sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
```

Comment:
Is this separator safe to use?

What happens if my repository is called `dave_tucker`...
Also what happens if the digest is not a SHA256 - can we reconstruct it in the same way?

Reply:
Yeah it's still safe to use, in that case it would just look like `dave_tucker_repo_sha256:fffff`

> Also what happens if the digest is not a SHA256 - can we reconstruct it in the same way?

This function just ensure we can go from IMAGE_URL to database key, as long as the digest is represented as a string we should always be fine.